### PR TITLE
`Development`: Fix flaky StructuralTestCaseServiceTest

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x]  This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] The changed server tests succeed **locally** and on **Bamboo** and on **GitHub Actions**.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
Multiple tests in StructuralTestCaseServiceTest were flaky. The tests frequently failed because of a Files.walk method that failed with a `NoSuchFileException`. The file that could not be found is called gc.log.lock, indicating that _git gc_ is still enabled, even though disabled in the `GitService` class.

### Description
In GitService, there is already an attempt to disable _auto gc_:
`gitRepoConfig.setInt(ConfigConstants.CONFIG_GC_SECTION, null, ConfigConstants.CONFIG_KEY_AUTO, 0);` 
However, this is not enough and more options need to be set to prevent the automatic execution of `git gc`.
Multiple options where taken from a helper method in `ProgrammingExerciseTestService`, disabling auto gc.

MySQL run without failing StructuralTestCaseServiceTest(s):
https://github.com/ls1intum/Artemis/actions/runs/6504427907/job/17666361382?pr=7328

Postgres run without failing StructuralTestCaseServiceTest(s):
https://github.com/ls1intum/Artemis/actions/runs/6504427907/job/17666361159?pr=7328

### Steps for Testing
_Although the behaviour should not have changed. I'd still like two reviewers to test Programming Exercises._
1. Create Programming Exercises, make sure the template and solution build run as expected.
2. As a student, participate in a newly created Programming Exercise (locally and on a test server).
3. As a student, participate in a newly created Programming Exercise in an Exam.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by b-fein — DISMISSED** [review]

File: src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java [hunk 1]
```diff
@@ -606,14 +606,19 @@ public Repository getExistingCheckedOutRepositoryByLocalPath(@NotNull Path local
...
+            gitRepoConfig.setBoolean(ConfigConstants.CONFIG_RECEIVE_SECTION, null, ConfigConstants.CONFIG_KEY_AUTOGC, false);
+
+            // disable symlinks to avoid security issues such as remote code execution
             gitRepoConfig.setBoolean(ConfigConstants.CONFIG_CORE_SECTION, null, ConfigConstants.CONFIG_KEY_SYMLINKS, false);
             gitRepoConfig.setBoolean(ConfigConstants.CONFIG_COMMIT_SECTION, null, ConfigConstants.CONFIG_KEY_GPGSIGN, false);
             gitRepoConfig.setString(ConfigConstants.CONFIG_BRANCH_SECTION, defaultBranch, ConfigConstants.CONFIG_REMOTE_SECTION, REMOTE_NAME);
             gitRepoConfig.setString(ConfigConstants.CONFIG_BRANCH_SECTION, defaultBranch, ConfigConstants.CONFIG_MERGE_SECTION, "refs/heads/" + defaultBranch);
 
-            // disable symlinks to avoid security issues such as remote code execution
             gitRepoConfig.save();
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java:622]: This method is quite long, consider extracting this part into its own method. Also double check if there are other places where this configuration needs to be applied.

**Comment by JanaNF** [comment]
Manual tested on ts2 with the the user artemis_test_user_1 (roll: Student)

1. Creating an exercise worked as aspected.
2. I could not start newly created or existing programming exercises. I can start quizzes or other exercises like modeling or text. 

The error message is:
image

File: src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java (hunk 1 above)
Comment (DominikRemo) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java]: I extracted a method for creating the new Repository.

I've searched through the code again but this is the only place (except tests) where we set the repo configs.

**Comment by DominikRemo** [comment]
> Manual tested on ts2 with the the user artemis_test_user_1 (roll: Student)
> 
> 1. Creating an exercise worked as aspected.
> 2. I could not start newly created or existing programming exercises. I can start quizzes or other exercises like modeling or text.

@JanaNF, thank you for testing this PR. Could you please have another look on another test server or with another test user. There currently seems to be a problem with artemis_test_user_1 on ts2.

**Review by Strohgelaender — APPROVED** [review]
code

**Review by laadvo — APPROVED** [review]
Tested on ts2, programming exercises can still be created. Student participation works for course and exam programming exercises :+1: 
Code also looks good.

**Review by b-fein — APPROVED** [review]

**Review by JanaNF — APPROVED** [review]
Manual tested on ts1. Worked as expected

**Review by krusche — APPROVED** [review]

**Pull request merged**